portal/models/questionnaire_response.py

```diff
@@ -906,3 +919,33 @@ def author_role(row_data, qnr_data):
                         row_data['option_text'] = strip_tags(
                             answer['valueCoding'].get('text', None))
                 yield {k: v for k, v in row_data.items() if v is not None}
+
+
+def first_last_like_qnr(qnr):
+    """Specialized lookup function to return similar QNRs
+
+    :param qnr: reference QNR - look for first and most recent QNRs similar
+      to the one provided.  That is owned by the same subject and QB.
```

Comment:
@pbugni  Since we have two QB's for the EMPRO patient questionnaire now (baseline and recurring), don't you need a TODO here to accommodate?

Reply:
d'OH!  right you are, I'll refactor that to fit.

Reply:
resolved via https://github.com/uwcirg/truenth-portal/pull/3814/commits/8fbef49d0fac2919324c1ea55552ed3a9b7ec593